torch/_inductor/lowering.py

```diff
@@ -446,14 +446,24 @@ def inner(*inputs: List[TensorBox], alpha=None):
                 other.get_size()
             ), f"ndim mismatch {fn} {ranges} {other.get_size()}"
 
+        emulate_precision_casts = (
+            V.graph.current_node is not None
+            and V.graph.current_node.meta is not None
+            and V.graph.current_node.meta.get("low_precision_pointwise_barrier", False)
+        )
+
         def inner_fn(index):
             assert len(index) == len(ranges), f"wrong ndim {index} {ranges}"
             if dtype == torch.bool and override_fn_when_input_bool is not None:
                 return override_fn_when_input_bool(*[load(index) for load in loaders])
             elif override_fn_when_cuda_float64 and is_cuda and dtype == torch.float64:
                 return override_fn_when_cuda_float64(*[load(index) for load in loaders])
             else:
-                return fn(*[load(index) for load in loaders])
+                out = fn(*[load(index) for load in loaders])
+                if emulate_precision_casts and dtype in (torch.bfloat16, torch.float16):
+                    downcast = ops.to_dtype(out, dtype, use_compute_types=False)
+                    return ops.to_dtype(downcast, dtype)
```

Comment:
Worth a comment explaining that this `do_dtype` actually cast the input to fp32.

Reply:
although `dtype` is fp16/bf16 here